src/odemis/gui/cont/tabs.py

```diff
@@ -3918,78 +3932,219 @@ def _onClickFocus(self, evt):
+        Set the focus detectors combobox selection to the the stream's detector
+        :param stream:
+        :param updated:
+        :return:
+        """
+        # Pick the stream to be shown:
+        #  1. Pick the stream which is playing (should_update=True)
+        #  2. Pick the focus stream which is already shown (in v.getStreams())
+        #  3. Pick the first focus stream
+        focusedview = self.tab_data_model.focussedView.value
+        if updated:
+            # This stream should be playing, remove the other ones first
+            for st in focusedview.getStreams():
+                if st in self._focus_streams:
+                    focusedview.removeStream(st)
+            focusedview.addStream(stream)
+            # Set the focus detectors combobox selection
+            istream = self._focus_streams.index(stream)
+            if istream != -1:
+                self.panel.cmb_focus_detectors.SetSelection(istream)
+        else:
+            # Check if there are any focus stream currently playing
+            current_stream = next((s for s in focusedview.getStreams() if s in self._focus_streams), None)
+            # Otherwise play the first one
+            if not current_stream:
+                focusedview.addStream(self._focus_streams[0])
+                self.panel.cmb_focus_detectors.SetSelection(0)
+
+    def _onFocusDetectorChange(self, evt):
+        """
+        Handler for focus detector combobox selection change
+        Play the stream associated with the chosen detector
+        """
+        if not self.panel.cmb_focus_detectors:
+            return
+        detector_name = self.panel.cmb_focus_detectors.GetValue()
+        # Find the stream related to this selected detector
+        stream = next((s for s in self._focus_streams if s.detector.name == detector_name), None)
+        if not stream:
+            logging.warning("Selected detector doesn't have a corresponding stream!")
+            return
+        # Play this stream (set both should_update and is_active with True)
+        stream.should_update.value = True
+        stream.is_active.value = True
+
+    def add_combobox_control(self, label_text, value=None, conf=None):
+        """ Add a combo box to the focus panel
+        # HACK to be used to automatically bind gratings to the combobox in the focus panel
+        """
+        # No need to create components, defined already on the xrc file
+        return self.panel.cmb_focus_gratings_label, self.panel.cmb_focus_gratings
+
+    def _enableManualFocusComponents(self):
+        """
+        Enable/Disable manual focus GUI components (pos slider, detector combobox, grating combobox)
+        Enable/Disable autofocus button
+        Enable/Disable main stream panel
+        """
+        # Ignore streak align mode..
+        align_mode = self.tab_data_model.align_mode.value
+        if align_mode == "streak-align":
+            return
```

Comment:
It's taken care of by `self.panel.pnl_focus.Enable(False)` but this check is still needed since both lens and streak focus go through the same manual focus function. 
Moved it to the caller.